indexer/src/main/java/au/org/aodn/esindexer/service/StacCollectionMapperService.java

```diff
@@ -145,25 +143,65 @@ List<String[]> createExtentTemporal(MDMetadataType source) {
         }
         return result;
     }
-
-    private String convertDateToZonedDateTime(String uuid, String inputDateString) {
-
-        String inputDateTimeString = inputDateString;
-        if (!inputDateString.contains("T")) {
-            inputDateTimeString += "T00:00:00";
-        }
-
+    /**
+     * If the date missing month / day / time then we will add it back by making it cover a range that is as wide as
+     * possible. So for example if only year then it will be first date of year and end date of that year.
+     *
+     * @param uuid - The uuid of the record
+     * @param dateStr - The date value in the XML
+     * @param isStartDate - Is it processing start date?
+     * @return - Well format date time string
+     */
+    private String convertDateToZonedDateTime(String uuid, String dateStr, boolean isStartDate) {
+        ZonedDateTime utcZonedDateTime = null;
+        String convertedDateTime = null;
         try {
-            ZonedDateTime zonedDateTime = ZonedDateTime.parse(inputDateTimeString, TemporalUtils.TIME_FORMATTER.withZone(ZoneId.of(timeZoneId)));
+            // Case 1: Date and Time (e.g., "2024-09-10T10:15:30")
+            if (dateStr.matches("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}")) {
+                // Do nothing
+                convertedDateTime = dateStr;
+                ZonedDateTime zt = ZonedDateTime.parse(convertedDateTime, TemporalUtils.TIME_FORMATTER.withZone(ZoneId.of(timeZoneId)));
```

Comment:
`ZonedDateTime zt = ZonedDateTime.parse(convertedDateTime, TemporalUtils.TIME_FORMATTER.withZone(ZoneId.of(timeZoneId)));` repeated twice, is that necessary?

Reply:
line 179 & 189, `ZonedDateTime zt =  ld.atZone(ZoneId.of(timeZoneId));`

so can it become ...

```java
ZonedDateTime zt;
if .... {
  zt = localDateTime.atZone(ZoneId.of(timeZoneId));
} else {
  zt = ZonedDateTime.parse(convertedDateTime, TemporalUtils.TIME_FORMATTER.withZone(ZoneId.of(timeZoneId)));
}
---

other than that, looks good to me